src/main/java/com/impactupgrade/nucleus/client/SfdcClient.java

```diff
@@ -646,6 +646,92 @@ protected QueryResult querySmsContacts(String updatedSinceClause, String filter,
     return query(query);
   }
 
+  //TODO: getDonators? getDonatorContacts?
+  public List<QueryResult> getDonatingContacts(Calendar updatedSince)
+      throws ConnectionException, InterruptedException {
+    List<QueryResult> queryResults = new ArrayList<>();
+
+    String updatedSinceClause = "";
+    if (updatedSince != null) {
+      updatedSinceClause = "SystemModStamp >= " + new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").format(updatedSince.getTime());
+    }
+    queryResults.add(queryDonatingContacts(updatedSinceClause));
+    queryResults.add(queryDonatingAccountsContacts(updatedSinceClause));
+
+    if (updatedSince != null) {
+      updatedSinceClause = "Id IN (SELECT ContactId FROM CampaignMember WHERE SystemModStamp >= " + new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").format(updatedSince.getTime()) + ")";
+      queryResults.add(queryDonatingContacts(updatedSinceClause));
+    }
+
+    return queryResults;
+  }
+
+  protected QueryResult queryDonatingContacts(String updatedSinceClause) throws ConnectionException, InterruptedException {
+    if (Strings.isNullOrEmpty(updatedSinceClause)) {
+      env.logJobWarn("no filter provided; out of caution, skipping the query to protect API limits");
+      return new QueryResult();
+    }
+    String updatedContactsQuery = "SELECT Id FROM Contact " +
+        "WHERE " + updatedSinceClause;
+    List<SObject> updatedContacts = queryListAutoPaged(updatedContactsQuery);
+    if (CollectionUtils.isEmpty(updatedContacts)) {
+      return new QueryResult();
+    }
+    String updatedContactIds = updatedContacts.stream()
+        .map(sObject -> "'" + sObject.getField("Id") + "'")
+        .collect(Collectors.joining(","));
+
+    String totalOpportunityAmountQuery = "SELECT ContactId, SUM(Amount) TotalAmount FROM Opportunity " +
+        "WHERE ContactId IN (" + updatedContactIds + ")" +
+        "GROUP BY ContactId " +
+        "HAVING SUM(Amount) > 0";
```

Comment:
Instead of this separate query, you can technically do this in the original query. Something like:

`SELECT [contact fields] FROM Contact WHERE [other clauses] AND Account.npo02__TotalOppAmount__c > 0.0`

That will give you all contacts that are within a household or business that has donated. The vast majority of them are single-member households. Using the Account total opp amount is generally more trustable than using the total opp amount on the Contact object.